### PR TITLE
Add Automated Testing on the DevOps Development Process file

### PR DESCRIPTION
**Pull request by GreatArcStudios**
The addition are for the automated testing on the devops portion of the `Development_Process.md` file. It goes over background of automated testing and various frameworks for unit testing and integration testing.

### CODE REVIEW

File: Topics/Development_Process.md [hunk 1]
```diff
@@ -21,3 +21,74 @@ The SOLID principles are:
...
+Further, there are usually "pipelines" for each project (e.g.,
+[AWS CodePipeline](https://docs.aws.amazon.com/codepipeline/latest/userguide/welcome-introducing.html)),
+and each of your code commits "flows" through your project pipeline. Generally, pipelines are
+usually composed of a series of "stages" that are executed sequentially. For example, after some
+cloud machine builds your code, the first testing stage might be to run the unit tests, and the
+second testing stage might be to run the integration tests. These setups vary from project to
+project, and you may be able to override passing criteria, but testing can lead credence to your
+implementation, and can support allow your code to promote (pass on to) production.
+
+![AWS CodePipeline Diagram](https://docs.aws.amazon.com/images/codepipeline/latest/userguide/images/pipeline-elements-workflow-application.png){fig-align="center"}
```
Comment (lmg8) [inline Topics/Development_Process.md:54]: `{fig-align="center"}` does not seem to be aligning on my end:

image

File: Topics/Development_Process.md [hunk 2]
```diff
@@ -21,3 +21,74 @@ The SOLID principles are:
...
+For example, Python uses [unittest](https://docs.python.org/3/library/unittest.html), Java uses
+[JUnit](https://junit.org/junit5/) or [Mockito](https://site.mockito.org/), and
+JavaScript/TypeScript uses [Jest](https://jestjs.io/). A good choice is one that supports mocking,
+spies, and other features that allow you to avoid testing 3rd party function calls (don't run them).
+Instead you should assume some expected behaviour from 3rd party function calls, and write tests
+that cover every "branch" of your code (code coverage). With that stated, some resources for the
+most important parts of unit testing are:
+
+-   **Test Coverage**:
+    - Altassian: [What is Test Coverage?](https://www.atlassian.com/continuous-delivery/software-testing/code-coverage)
```
Comment (lmg8) [inline Topics/Development_Process.md:71]:
```suggestion
    - Atlassian: [What is Test Coverage?](https://www.atlassian.com/continuous-delivery/software-testing/code-coverage)
```

**Review by lmg8 — CHANGES_REQUESTED** [review]
These are some very helpful resources for testing! I would consider creating a separate page similar to what's already been done. Maybe a section for testing so we do not overcrowd the Development Process page.  

image

**Comment by GreatArcStudios** [comment]
Re new section: Could you expand on that, and perhaps show an example in markdown of what you mean? I am following the format of the assignment/readme. I am under the impression that the prof wants the structure to adhere to:

image

File: Topics/Development_Process.md (hunk 1 above)
Comment (GreatArcStudios) [reply Topics/Development_Process.md:54]: Removed

File: Topics/Development_Process.md (hunk 2 above)
Comment (GreatArcStudios) [reply Topics/Development_Process.md:71]: Fixed

**Review by lmg8 — COMMENTED** [review]
You can see it [Development_Process.md](https://github.com/learning-software-engineering/learning-software-engineering.github.io/blob/main/Topics/Development_Process.md)
how other students created separate file and linked it to the main Development Process page. Feel free to ask more questions if you're still not sure.

Here's an example PR of someone adding a separate Django page in Development Process:
https://github.com/learning-software-engineering/learning-software-engineering.github.io/pull/19/files

**Comment by GreatArcStudios** [comment]
Changed

**Review by lmg8 — APPROVED** [review]
Looks good! Thank you 👍

**Comment by lmg8** [comment]
@GreatArcStudios it looks like Automated Testing is under build requirements rather than resources for development process:
Can you create a new PR to fix this? Tag me on the new PR and I will merge

image

**Pull request merged**